main.js

```diff
@@ -0,0 +1,298 @@
+  startButton1.disabled = true
+  xName.value = ""
+  oName.value = ""
+  let time0 = performance.now()
+  twoPlayerGame()
+  time1 = performance.now()
+  startButton2.innerHTML = "Time Elapsed: " + (Math.round(time1 - time0)) + " seconds" // this is meant to be my timer, but it measures how long it takes to click the start button. I think this setup will work, but I don't know where to put the start and end points for the timer.
+}
+
+// game proper (with switching players)
+
+let twoPlayerGame = function () {
+
+  // add click event to each cell on the grid
+
+  for (let square of cells) {
+    square.addEventListener('click', gaming)
+
+    function gaming() {
+
+      // prevents clicking on a filled square by throwing alert 
+      if (square.innerHTML !== "") {
+        window.alert("Please choose an empty square!")
+      } else {
+        square.innerHTML = currentPlayer.symbol;
+      }
+
+      // checks for tie       
+      tieGame()
+
+      // alternating X / O turns, checking for winning combinations
+
+      if (square.innerHTML === "X") {
+        gameWin()
+
+        if (checkedMoves.length === 3) {
+          playerStatus.innerHTML = currentPlayer.name + " wins!"
+          delayEnding()
+        } else if (fullGrid.length === 9) {
+          playerStatus.innerHTML = "It's a draw!"
+          delayEnding()
+        }
+        else {
+          currentPlayer = playerO
+          playerStatus.innerHTML = currentPlayer.name + "'s Turn";
+        }
+
+      } else {
+        gameWin()
+
+        if (checkedMoves.length === 3) {
+          playerStatus.innerHTML = currentPlayer.name + ' wins!'
+          delayEnding()
+        } else if (fullGrid.length === 9) {
+          playerStatus.innerHTML = "It's a draw!"
+          delayEnding()
+        }
+        else {
+          currentPlayer = playerX
+          playerStatus.innerHTML = currentPlayer.name + "'s Turn";
+        }
+
+      }
+    }
+  }
+}
+
+// ------------------------------------------------------------------------------------------------------------
+// one-player game between human and computer
+
+// starts game, disabling both player option buttons 
+
+function startGame1() {
+  playerX.name = "Player X"
+  playerO.name = "Player O"
+  currentPlayer = playerX
+  playerStatus.innerHTML = currentPlayer.name + "'s Turn";
+  event.target.disabled = true
+  startButton2.disabled = true
+  onePlayerGame()
+}
+
+// random number generator to choose among the available spaces in the board array (by index)
+
+function randomInt(max, min) {
+  max = board.length - 1
+  min = 0
+  arrayIndex = Math.floor(min + (Math.random() * (max - min + 1)))
+  return arrayIndex
+}
+
+// main game function 
+
+board = [c0, c1, c2, c3, c4, c5, c6, c7, c8]
+
+function onePlayerGame() {
+
+  for (let square of cells) {
+    square.addEventListener('click', playGame)
+
+    // functions that use the "square" variable which (due to scope) cannot be defined outside of the onePlayerGame
```

Comment:
There is a way to define these functions outside of the onePlayerGame() function. When you add an event listener to an element on the DOM, the 'event' is automatically passed into the callback function being used. So instead of using the 'square' variable which corresponds to an element, you can use 'event.target', which also corresponds to the element where the event came from.